.github/workflows/code-quality.yml

```
@@ -0,0 +1,254 @@
+          echo "Running markdownlint on Markdown files..."
+          markdownlint "**/*.md" --ignore node_modules || true
+          echo "Markdown linting completed"
+
+  yaml-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
```

actionlint (1.7.7)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)